crates/turbopack-ecmascript/src/analyzer/mod.rs

```diff
@@ -1051,6 +1178,156 @@ impl JsValue {
             | JsValue::FreeVar(_) => true,
         }
     }
+
+    pub fn is_truthy(&self) -> Option<bool> {
```

Comment:
`Option<bool>` serves the role of a ternary boolean. It would be clearer to use an enum and simplify your logic functions:

```
pub enum Ternary {
    True,
    False,
    Unknown,
}

impl Ternary {
    pub fn is_true() -> bool;
    pub fn is_false() -> bool;
    pub fn is_unknown() -> bool;
}
```

Reply:
It makes the `match`es and other operations more clear what they're operating on. `merge_if_known`, `all_if_known`, and `shortcircut_if_known` would also read more easily.

Reply:
What should I do with the `Option<&str>` of `as_str`? It's also string or unknown...?

Adding Ternary makes it a bit inconsistent. I probably need a `MaybeKnown<T>`?

Reply:
Let's not block on this. I don't have a strong opinion on this.